add tbon.interface-hint broker attribute / configuration key with CIDR network support

garlick opened the pull request:
Problem:  starting an instance with its overlay assigned to a particular network is tricky if you network interface names are unpredictable.  And environment variables are a hacky way to influence the broker's choice.

The broker chooses a network address to bind to (and to exchange with PMI peers) using a heuristic usually based on the default route.  You can set environment variables to choose the address associated with the hostname instead, or a specific interface by name.

This adds
- the option of specifying the interface by network address (ipv4  CIDR, e.g. 10.0.2.0/24)
- a new broker attribute `tbon.interface-hint` which can have the values of `default-route` (the default), `hostname` (as described above), or name that can be either an interface name or a network address
- a TOML config key of the same name

Fixes #5754

grondo reviewed (approved):
LGTM!

Unrelated to this PR (reproduced on current master), if the broker is unable to find a matching ip addr (e.g. no addr matches the provided CIDR) an error is printed, but the instance hangs until the doom plugin timer goes off:

```
lt-flux-broker: could not find address associated with 192.168.1.1/24
lt-flux-broker: bootstrap failed
flux-shell[0]: FATAL: doom: rank 0 on host corona211 exited and exit-timeout=30s has expired
```

If an instance was being launched with `exit-timeout=none`, then I assume this would result in a hang until the allocation expired.

garlick commented:
Thanks, fixed that style issue and rebased on current master.

Good catch on the hang.   It seems like the broker should be calling PMI_Abort() in that case but the upmi abstraction is not yet plumbed for it, so it's probably a separate PR.  I opened #6278 on that one.

Will set MWP.

garlick commented:
> One other thought here: is there any way to have an address hint that says "same address as parent"?

Oh interesting.  We could even pre-populate the interface as a key in the PMI kvs.  Good idea!
